pkg/generator/helper/helper.go

```diff
@@ -0,0 +1,308 @@
+	{{- range $key, $value := .Metadata }}
+	cmd.Flags().{{ $value.Type | flagType }}(&my{{ $.StepName | title }}Options.{{ $value.Name | golangName }}, "{{ $value.Name }}", {{ $value.Default }}, "{{ $value.Description }}"){{ end }}
+	{{- printf "\n" }}
+	{{- range $key, $value := .Metadata }}{{ if $value.Mandatory }}
+	cmd.MarkFlagRequired("{{ $value.Name }}"){{ end }}{{ end }}
+}
+
+// retrieve step metadata
+func {{ .StepName }}Metadata() config.StepData {
+	var theMetaData = config.StepData{
+		Spec: config.StepSpec{
+			Inputs: config.StepInputs{
+				Parameters: []config.StepParameters{
+					{{- range $key, $value := .Metadata }}
+					{
+						Name:      "{{ $value.Name }}",
+						Scope:     []string{{ "{" }}{{ range $notused, $scope := $value.Scope }}"{{ $scope }}",{{ end }}{{ "}" }},
+						Type:      "{{ $value.Type }}",
+						Mandatory: {{ $value.Mandatory }},
+					},{{ end }}
+				},
+			},
+		},
+	}
+	return theMetaData
+}
+`
+
+//StepTestGoTemplate ...
+const stepTestGoTemplate = `package cmd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test{{.CobraCmdFuncName}}(t *testing.T) {
+
+	testCmd := {{.CobraCmdFuncName}}()
+
+	// only high level testing performed - details are tested in step generation procudure
+	assert.Equal(t, "{{.StepName}}", testCmd.Use, "command name incorrect")
+
+}
+`
+
+// ProcessMetaFiles generates step coding based on step configuration provided in yaml files
+func ProcessMetaFiles(metadataFiles []string, openFile func(s string) (io.ReadCloser, error), writeFile func(filename string, data []byte, perm os.FileMode) error, exportPrefix string) error {
+	for key := range metadataFiles {
+
+		var stepData config.StepData
+
+		configFilePath := metadataFiles[key]
+
+		metadataFile, err := openFile(configFilePath)
+		checkError(err)
+		defer metadataFile.Close()
+
+		fmt.Printf("Reading file %v\n", configFilePath)
+
+		err = stepData.ReadPipelineStepData(metadataFile)
+		checkError(err)
+
+		fmt.Printf("Step name: %v\n", stepData.Metadata.Name)
+
+		osImport := false
+		osImport, err = setDefaultParameters(&stepData)
+		checkError(err)
+
+		myStepInfo := getStepInfo(&stepData, osImport, exportPrefix)
+
+		step := stepTemplate(myStepInfo)
+		err = writeFile(fmt.Sprintf("cmd/%v_generated.go", stepData.Metadata.Name), step, 0644)
+		checkError(err)
+
+		test := stepTestTemplate(myStepInfo)
+		err = writeFile(fmt.Sprintf("cmd/%v_generated_test.go", stepData.Metadata.Name), test, 0644)
+		checkError(err)
+	}
+	return nil
+}
+
+func openMetaFile(name string) (io.ReadCloser, error) {
+	return os.Open(name)
+}
+
+func fileWriter(filename string, data []byte, perm os.FileMode) error {
+	return ioutil.WriteFile(filename, data, perm)
+}
+
+func setDefaultParameters(stepData *config.StepData) (bool, error) {
+	//ToDo: custom function for default handling, support all relevant parameter types
+	osImportRequired := false
+	for k, param := range stepData.Spec.Inputs.Parameters {
```

Comment:
@OliverNocon must be extended to handle conditions and filter duplicate conditional defaults of the same parameter.

Reply:
shall we do this in this PR or rather in another one?
So far the generator did not cover this, right?

Reply:
Could also be handled in a separate PR. Was more meant to bring it up as a topic;-)

Reply:
ok, then let's keep this in mind an put this in separately.
We definitely need to take care about this!